src/box/lua/net_box.c

```diff
@@ -2166,8 +2193,18 @@ luaT_netbox_transport_perform_async_request(struct lua_State *L)
 {
 	struct netbox_transport *transport = luaT_check_netbox_transport(L, 1);
 	struct netbox_request *request = lua_newuserdata(L, sizeof(*request));
-	if (luaT_netbox_transport_make_request(L, 2, transport, request) != 0)
-		return luaT_push_nil_and_error(L);
+	if (luaT_netbox_transport_make_request(L, 2, transport, request) != 0) {
+		enum netbox_method method = lua_tointeger(L, 9);
```

Comment:
`method` is already stored in `request->method`.

Reply:
It's not. `luaT_netbox_transport_make_request` will return non zero in case there's an error occured. And all checks for errors are done before the initialization of the request object. So request will be initialized to this point